### DISCUSSION TITLE
How to modify existing vim keybinds?

### DISCUSSION THREAD

**Discussion by rakim0** [zed-industries/zed#20790] [Feature Requests]
I couldn't figure out to change the keybinds. Any help?

From the docs I was able to reach this point but couldn't figure how to remap ctrl-d to ctrl-d+zz
```
    {
        "context": "VimControl && !menu",
        "bindings": {
            // Put key bindings here if you want them to work in normal & visual mode.
            "ctrl-d": <command>
        }
    },
```
Trying to achieve similar results to adding this in vim

```
nnoremap <C-d> <C-d>zz
nnoremap <C-u> <C-u>zz
```

**Comment by 0x2CA** [comment]
```json
 {
    "context": "VimControl && !menu",
    "bindings": {
      "vimScrollDown": "vim::PageDown",
      "vimScrollUp": "vim::PageUp",
      "ctrl-d": ["workspace::SendKeystrokes", "vimScrollDown z z"],
      "ctrl-u": ["workspace::SendKeystrokes", "vimScrollUp z z"]
    }
 }
```

**Reply by rakim0** [reply]
Thank you! Could you share how you found out about `vim::PageDown`?

**Reply by 0x2CA** [reply]
> Thank you! Could you share how you found out about `vim::PageDown`?

![image](https://github.com/user-attachments/assets/eed85a85-1824-479e-a0e2-8390ee88224d)
![image](https://github.com/user-attachments/assets/67818231-e617-4fa7-b78e-e567d090d0d9)

**Reply by rakim0** [reply]
Ah I was trying to look for this in the zed:default keymap. Didn't know there was vim:default keymap too. 
Thanks again!

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/support
